ibm/resource_ibm_private_dns_permitted_network.go

```diff
@@ -0,0 +1,193 @@
+package ibm
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	pdnsVpcCRN                     = "vpc_crn"
+	pdnsNetworkType                = "type"
+	pdnsPermittedNetworkID         = "permitted_network_id"
+	pdnsPermittedNetworkCreatedOn  = "created_on"
+	pdnsPermittedNetworkModifiedOn = "modified_on"
+	pdnsPermittedNetworkState      = "state"
+)
+
+var allowedNetworkTypes = []string{
+	"vpc",
+}
+
+func resourceIBMPrivateDNSPermittedNetwork() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMPrivateDnsPermittedNetworkCreate,
+		Read:     resourceIBMPrivateDnsPermittedNetworkRead,
+		Delete:   resourceIBMPrivateDnsPermittedNetworkDelete,
+		Exists:   resourceIBMPrivateDnsPermittedNetworkExists,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			pdnsPermittedNetworkID: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			pdnsInstanceID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			pdnsZoneID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			pdnsNetworkType: {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
Instead of Required...Make it optional and Default: "vpc" and t validateFunc can use existing method
ValidateFunc: validateAllowedStringValue([]string{"vpc"
}),

Reply:
addressed.